src/js/common/containers/DowntimeNotification.jsx

```diff
@@ -0,0 +1,248 @@
+  arcgis: 'arcgis',
+  dslogon: 'dslogon',
+  emis: 'emis',
+  es: 'es',
+  evss: 'evss',
+  idme: 'idme',
+  mvi: 'mvi',
+  mhv: 'mhv',
+  tims: 'tims',
+  vic: 'vic'
+};
+
+export const serviceStatus = {
+  down: 'down',
+  downtimeApproaching: 'downtimeApproaching',
+  ok: 'ok'
+};
+
+function DowntimeNotificationWrapper({ status, children }) {
+  return <div id="downtime-notification" data-status={status} style={{ marginBottom: '1em' }} className="row-padded">{children}</div>;
+}
+
+class DowntimeNotification extends React.Component {
+
+  static propTypes = {
+    appTitle: PropTypes.string,
+    children: PropTypes.node,
+    content: PropTypes.node,
+    dependencies: PropTypes.arrayOf(PropTypes.oneOf(objectValues(services))).isRequired,
+    determineStatus: PropTypes.func,
+    getScheduledDowntime: PropTypes.func.isRequired,
+    isReady: PropTypes.bool,
+    loadingIndicator: PropTypes.node,
+    render: PropTypes.func,
+    scheduledDowntime: PropTypes.arrayOf(
+      PropTypes.shape({
+        service: PropTypes.string,
+        description: PropTypes.string,
+        startTime: PropTypes.instanceOf(Date),
+        endTime: PropTypes.instanceOf(Date)
+      })
+    ),
+    userIsAuthenticated: PropTypes.bool
+  };
+
+  static defaultProps = {
+    appTitle: 'application'
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalDismissed: false,
+      cache: {}
+    };
+  }
+
+  componentWillMount() {
+    this.props.getScheduledDowntime();
+  }
+
+  // This is here just for caching the result of calculateDowntime, so that it isn't run every time a prop changes.
+  // Currently, this component should only do its calculations once, because it would be weird if in the middle of filling
+  // out a form a modal or alert appears alerting them of downtime, or if it just shuts down the app because we entered
+  // scheduled downtime.
+  componentWillReceiveProps(newProps) {
+    const firstLoad = newProps.isReady && !this.props.isReady;
+    if (firstLoad) {
+      const downtimeMap = this.calculateDowntime(newProps.dependencies, newProps.scheduledDowntime);
+      const status = this.determineStatus(downtimeMap);
+      const downtimeWindow = this.getDowntimeWindow(downtimeMap[status]);
+      const cache = { downtimeMap, status, downtimeWindow };
+      this.setState({ cache });
+    }
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    if (!nextProps.isReady) return false;
+    if (nextProps.isReady && !this.props.isReady) return true;
+
+    return nextState.cache.status !== serviceStatus.down;
+  }
+
+  getStatusForDowntime(downtime, current = moment(), warning = moment().add(1, 'hour')) {
+    const inclusive = '[]';
+
+    if (!downtime) return serviceStatus.ok;
+    if (current.isBetween(downtime.startTime, downtime.endTime, inclusive)) return serviceStatus.down;
+    if (warning.isBetween(downtime.startTime, downtime.endTime, inclusive)) return serviceStatus.downtimeApproaching;
```

Comment:
I wrote this assuming that outages will be at least an hour long so I liked the explicitness of `isBetween` but otherwise I think you're right. @patrickvinograd can you confirm?